src/xrpld/shamap/detail/SHAMap.cpp

```diff
@@ -137,6 +137,17 @@ SHAMap::walkTowardsKey(uint256 const& id, SharedPtrNodeStack* stack) const
     auto inNode = root_;
     SHAMapNodeID nodeID;
 
+    // The item should definitely be there if it's not backed and
+    // inner->isEmptyBranch(branch) returns true, otherwise, the node may not be
+    // fully ready, and we should not throw an exception.
+    bool const shouldThrow = !backed_;
+
+    auto getDescend = [this, shouldThrow](SHAMapInnerNode& parent, int branch) {
```

Comment:
You do not need above line, if you initialize `shouldThrow` in capture list:

```suggestion
    auto getDescend = [this, shouldThrow = !backed_](SHAMapInnerNode& parent, int branch) {
```